doc/examples/uuid.rst

```diff
@@ -0,0 +1,472 @@
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import UuidRepresentation
+
+  # No configured UUID representation
+  collection = client.python_legacy.get_collection('test', codec_options=DEFAULT_CODEC_OPTIONS)
+
+  # Using UuidRepresentation.PYTHON_LEGACY
+  pylegacy_opts = CodecOptions(uuid_representation=UuidRepresentation.PYTHON_LEGACY)
+  pylegacy_collection = client.python_legacy.get_collection('test', codec_options=pylegacy_opts)
+
+  # UUIDs written by PyMongo with no UuidRepresentation configured can be queried using PYTHON_LEGACY
+  uuid_1 = uuid4()
+  collection.insert_one({'uuid': uuid_1})
+  document = pylegacy_collection.find_one({'uuid': uuid_1})
+
+  # UUIDs written using PYTHON_LEGACY can be read by PyMongo with no UuidRepresentation configured
+  uuid_2 = uuid4()
+  pylegacy_collection.insert_one({'uuid': uuid_2})
+  document = collection.find_one({'uuid': uuid_2})
+
+``PYTHON_LEGACY`` encodes native :class:`uuid.UUID` objects to
+:class:`~bson.binary.Binary` subtype 3 objects, preserving the same
+byte-order as :attr:`~uuid.UUID.bytes`::
+
+  from bson.binary import Binary
+
+  document = collection.find_one({'uuid': Binary(uuid_2.bytes, subtype=3)})
+  assert document['uuid'] == uuid_2
+
+.. _java-legacy-representation-details:
+
+``JAVA_LEGACY``
+^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used when reading UUIDs
+   written to MongoDB by the Java driver without an explicitly configured UUID
+   representation.
+
+The :data:`~bson.binary.UuidRepresentation.JAVA_LEGACY` representation
+corresponds to the legacy representation of UUIDs used by the MongoDB Java
+Driver.
+
+.. note:: The ``JAVA_LEGACY`` representation reverses the order of bytes 0-7,
+   and bytes 8-15.
+
+As an example, consider the same UUID described in :ref:`example-legacy-uuid`.
+Let us assume that an application used the Java driver without an explicitly
+specified UUID representation to insert the example UUID
+``00112233-4455-6677-8899-aabbccddeeff`` into MongoDB. If we try to read this
+value using PyMongo with no UUID representation specified, we end up with an
+entirely different UUID::
+
+  UUID('77665544-3322-1100-ffee-ddccbbaa9988')
+
+However, if we explicitly set the representation to
+:data:`~bson.binary.UuidRepresentation.JAVA_LEGACY`, we get the correct result::
+
+  UUID('00112233-4455-6677-8899-aabbccddeeff')
+
+PyMongo uses the specified UUID representation to reorder the BSON bytes and
+load them correctly. ``JAVA_LEGACY`` encodes native :class:`uuid.UUID` objects
+to :class:`~bson.binary.Binary` subtype 3 objects, while performing the same
+byte-reordering as the legacy Java driver's UUID to BSON encoder.
+
+.. _csharp-legacy-representation-details:
+
+``CSHARP_LEGACY``
+^^^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used when reading UUIDs
+   written to MongoDB by the C# driver without an explicitly configured UUID
```

Comment:
This wording makes it seem like all C# apps have this problem. Should we clarify that it's only legacy C# apps (which don't use STANDARD)?

Reply:
Done.